### PR TITLE
[JENKINS-33300] Upgrade to parent pom 2.3

### PR DESCRIPTION
**Pull request by armfergom**
@reviewbybees

* Upgraded to parent pom 2.3
* Fixed javadoc issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -53,78 +52,27 @@
...
     <plugins>
       <plugin>
-        <groupId>org.codehaus.mojo</groupId>
-        <artifactId>animal-sniffer-maven-plugin</artifactId>
-        <!-- details defined in the parent POM -->
-      </plugin>
-      <plugin>
         <groupId>org.codehaus.gmaven</groupId>
         <artifactId>gmaven-plugin</artifactId>
         <executions>
```
Comment (recena) [inline pom.xml]: Really needed?

Comment (armfergom) [inline pom.xml]: Yes, I think so. If removed, the version used will be a greater one and the plugin stops building due to an API incompatibility. Specifically, this one:

```
 [ERROR] Failed to execute goal org.codehaus.gmaven:gmaven-plugin:1.5-jenkins-3:testCompile (test-in-groovy) on project ldap: Execution test-in-groovy of goal org.codehaus.gmaven:gmaven-plugin:1.5-jenkins-3:testCompile failed: An API incompatibility was encountered while executing org.codehaus.gmaven:gmaven-plugin:1.5-jenkins-3:testCompile: java.lang.NoSuchMethodError: org.codehaus.groovy.ast.ModuleNode.getStarImports()Ljava/util/List;
```

**Comment by armfergom** [comment]
One more commit removing the `compatibleSinceVersion`configuration parameter for the maven-hpi-plugin. Not needed since parent pom version 1.321 as stated [here](https://wiki.jenkins-ci.org/display/JENKINS/Marking+a+new+plugin+version+as+incompatible+with+older+versions)

File: pom.xml [hunk 2]
```diff
@@ -53,103 +52,33 @@
...
-        <version>2.5</version>
       </plugin>
       <plugin>
         <groupId>org.jenkins-ci.tools</groupId>
         <artifactId>maven-hpi-plugin</artifactId>
-        <version>${maven-hpi-plugin.version}</version>
         <configuration>
-          <compatibleSinceVersion>1.9</compatibleSinceVersion>
           <systemProperties>
             <hudson.bundled.plugins>${basedir}/work/plugins/ldap.hpl</hudson.bundled.plugins>
```
Comment (andresrc) [inline pom.xml]: We should see if this is still needed, and if not remove the whole plugin entity.

Comment (armfergom) [inline pom.xml]: When the plugin is bundled, this is needed so that the HPI plugin uses the plugin being developed and not the one bundled. Same happens for example with the Subversion plugin.

File: src/main/java/hudson/security/LDAPSecurityRealm.java [hunk 3]
```diff
@@ -1291,7 +1291,7 @@ public String getValue() {
 
             @Override
             public String getDisplayName() {
-                return null;
+                return "";
```
Comment (amuniz) [inline src/main/java/hudson/security/LDAPSecurityRealm.java]: This seems to be not related to the parent pom upgrade. What's different on returning `null` or empty string? Is it fixing a bug?

Comment (armfergom) [inline src/main/java/hudson/security/LDAPSecurityRealm.java]: With the new parent pom, findbugs is integrated. With the current version of the core (1.566), the plugin builds ok and findbugs does not find issues.

However, if this is run against the latest core version, findbugs starts complaining because the method getDisplayName() in the superclass is annotated with @Nonnull. It is important to avoid this, so that the plugin compatibility test works correctly.

Comment (andresrc) [inline src/main/java/hudson/security/LDAPSecurityRealm.java]: @amuniz We are trying to get a clean  `mvn clean verify -Djenkins.version=last_lts` build as well when upgrading to the parent POM, to anticipate possible PCT issues.

**Comment by andresrc** [comment]
:bee:

**Comment by armfergom** [comment]
@reviewbybees done

**Comment by abayer** [comment]
:bee:, merging.
